packages/@react-aria/overlays/docs/usePopover.mdx

```diff
@@ -0,0 +1,278 @@
+
+* Exposes overlay trigger and connects trigger to overlay with ARIA
+* Positions the overlay relative to the trigger, automatically flipping when needed
+* Hides content behind the overlay from assistive technology while open
+* Handles closing the overlay when interacting outside and pressing the <Keyboard>Escape</Keyboard> key
+* Restores focus to the triggering element on unmount
+
+**Note**: `usePopover` only handles the overlay itself. It should be combined
+with [useDialog](useDialog.html) to create fully accessible popovers. Other overlays
+such as menus may also be placed in a popover.
+
+## Anatomy
+
+<Anatomy />
+
+A popover consists of a trigger element (e.g. button) and an overlay, which is positioned relative to the trigger. The overlay may contain a [dialog](useDialog.html), or another element such as a [menu](useMenu.html) or [listbox](useListBox.html) when used within a component such as a [select](useSelect.html) or [combobox](useComboBox.html).
+
+`usePopover` returns props that you should spread onto the trigger element and
+the appropriate element:
+
+<TypeContext.Provider value={docs.links}>
+  <InterfaceType properties={docs.links[docs.exports.usePopover.return.id].properties} />
+</TypeContext.Provider>
+
+State is managed by the <TypeLink links={statelyDocs.links} type={statelyDocs.exports.useOverlayTriggerState} />
+hook in `@react-stately/overlays`. The state object should be passed as an argument to `usePopover`.
+
+## Example
+
+This example shows how to build a typical popover overlay that is positioned relative to
+a trigger button. The content of the popover is a dialog, built
+with <TypeLink links={dialogDocs.links} type={dialogDocs.exports.useDialog} />. The `Dialog` component used in this example can also be reused within a [modal](useModalOverlay.html) or other types of overlays. The implementation is available below.
+
+The `Popover` component uses an &lt;<TypeLink links={docs.links} type={docs.exports.Overlay} />&gt; to render its contents in a React [Portal](https://reactjs.org/docs/portals.html) at the end of the document body, which ensures it is not clipped by other elements. It also acts as a focus scope, containing focus within the popover and restoring it to the trigger when it unmounts. <TypeLink links={docs.links} type={docs.exports.usePopover} /> handles positioning the popover relative to the trigger element, and closing it when the user interacts outside or presses the <Keyboard>Escape</Keyboard> key.
+
+<TypeLink links={docs.links} type={docs.exports.usePopover} /> also hides content outside the popover from screen readers, which is important since the surrounding content won't be in context of the original trigger due to the portal. To allow screen reader users to dismiss the popover without a keyboard (e.g. on mobile), visually hidden &lt;<TypeLink links={docs.links} type={docs.exports.DismissButton} />&gt; elements are added at the start and end of the popover.
```

Comment:
I'm not sure where the best place to put this would be, but perhaps there should be a small blurb explaining when to use the `isNonModal` prop (aka warning against setting it to true unless absolutely appropriate like in the combobox/searchautocomplete situation). IMO it isn't 100% clear from the prop description itself

Reply:
Added more detail to the prop description for now.